### ISSUE TITLE
Readme documents format?

### ISSUE THREAD

**Issue by matthew-dews** [mayaculpa/hapi#37]
The readme documents aren't in a format that github is able to display natively. Would it be possible to have them converted to Markdown files? That way new users can browse the documents from the web to better understand the workings of HAPI. It may be especially useful for users who are using binary releases and don't have the repo locally.

I'd be happy to do the conversion.

**Comment by PedroSFreitas** [comment]
Hey there @matthew-dews , **thanks for reaching out.**
Unfortunately those files are quite out of date. We are working to get them updated as soon as possible and we would be thrilled for your help.
Can you give us a few days?

**Comment by matthew-dews** [comment]
Sure thing, I'll keep an eye on the repository for new content as well.

**Comment by PedroSFreitas** [comment]
Great. **Thanks again!**
We'll keep you updated. :)

**Comment by james-prior** [comment]
markdown (.md) and restructured text (.rst) are great
for documentation that is both readable as plain text source
and renders to pretty stuff.

At this point, markdown is probably plenty sufficient.

**Comment by matthew-dews** [comment]
I agree, especially since Github plays nicely with them. Jekyll also is a great tool for building websites from Markdown files which I have done in the past with documentation. And Github lets you build Jekyll sites from subfolders within a repo (such as a docs folder) which can be a nice way to easily generate pretty documentation pages.

**Comment by ozarchie** [comment]
I have created https://hapidocs.github.io/ using "hapidocs at gmail.com" as
the github user. I am not expert at Jeckyll, so anyone who would like to
start the docs at that site is welcome.

**Comment by matthew-dews** [comment]
@PedroSFreitas If the old documentation is outdated and there's no up to date documentation in formats other than Markdown (e.g. the new README) then the issue can be closed. I'll be happy to jump on other issues as I run into them or they pop up.

**Comment by PedroSFreitas** [comment]
_** @ozarchie commented 13 minutes ago**_
> I have created https://hapidocs.github.io/ using "hapidocs at gmail.com" as
> the github user. I am not expert at Jeckyll, so anyone who would like to
> start the docs at that site is welcome.

That's nice @ozarchie, thanks a bunch. Maybe @matthew-dews could help you on that.

_** @matthew-dews commented 9 minutes ago**_
> @PedroSFreitas If the old documentation is outdated and there's no up to date documentation in formats other than Markdown (e.g. the new README) then the issue can be closed. I'll be happy to jump on other issues as I run into them or they pop up

I believe having a nice, well crafted, documentation outside our README, is a nice thing to have. We could then try to use (convert?) it as manual page on Linux (man). It's a new territory for me, so I might be saying something wrong here; let me know.
That said, maybe we should leave this Issue open until we have a good documentation outside the README. What do you think?

**_james-prior commented 3 hours ago_**
> At this point, markdown is probably plenty sufficient.

Agreed on "at this point".

Let's see if @TylerReedMC has something else to add to this conversation.

**Comment by matthew-dews** [comment]
@PedroSFreitas Sounds good on keeping the issue open for now.

Another option is https://readthedocs.org/ which relieves us of the need to worry about choosing and setting up a theme. In the past with Jekyll I have found the process of picking, editing and refining a theme was time consuming and tedious. One of the immediate benefits I see with readthedocs is that documentation can be versioned along with releases. They have [other features](https://docs.readthedocs.io/en/latest/features.html) as well. 
With Markdown/RST it is easy to switch between solutions but I'd like to see some direction from folks on the HAPI team saying lets go with X or Y before starting to really dive into something.

**Comment by james-prior** [comment]
When one starts talking about man pages and web sites,
I start leaning towards restructured text (rst) and sphinx.
rst because it supports fancier stuff than markdown
and sphinx because it can convert rst to many other formats.
But that is fancy stuff for later on.
Right now, we need basics.
I would not worry too much about markdown versus rst.
Terse accurate plain text is better than nothing.
[Perfect is the enemy of good](https://en.wikipedia.org/wiki/Perfect_is_the_enemy_of_good)

To paraphrase what I said on another issue or pull request:

> Working stuff beats pretty stuff.

Pedro added some sorely needed stuff in 1796a6d. Thanks Pedro!

[Release early, release often!](https://en.wikipedia.org/wiki/Release_early,_release_often)

**Comment by TylerReedMC** [comment]
Enayet and I were recently discussing adding API doc generation as a CI step on the new Jenkins server. We could use Sphinx for that. It would update with each new merge.

**Comment by TylerReedMC** [comment]
I've also added a Features page to the Wiki and updated the Scheduler page. This is really just an attempt to start dumping whatever relevant and existing documentation I have into github so we can all see it and decide how to handle.  -T

**Comment by TylerReedMC** [comment]
[Technologies.pdf](https://github.com/mayaculpa/hapi/files/1018759/Technologies.pdf)
[FMS Requirements.pdf](https://github.com/mayaculpa/hapi/files/1018765/FMS.Requirements.pdf)
[HAPI Enclosure Requirements.pdf](https://github.com/mayaculpa/hapi/files/1018766/HAPI.Enclosure.Requirements.pdf)

**Comment by TylerReedMC** [comment]
[HAPI Architecture - Simple Overview.pdf](https://github.com/mayaculpa/hapi/files/1018786/HAPI.Architecture.-.Simple.Overview.pdf)

**Comment by PedroSFreitas** [comment]
**@matthew-dews commented**
> Another option is https://readthedocs.org/ which relieves us of the need to worry about choosing and setting up a theme

That is definitely something nice for the future. Thanks!
Hey if you want to chat about something you can hit me on IRC (Freenode) by the nick `psf`. Or even take a look at our Mattermost (Open Source Slack alternative) platform here: http://104.131.99.114:8065/signup_user_complete/?id=hhyq3c9b4jd9d8dyb734ai894h

**@james-prior commented**
> When one starts talking about man pages and web sites,
> I start leaning towards restructured text (rst) and sphinx.

Noted! :D

**@TylerReedMC commented**
> Enayet and I were recently discussing adding API doc generation as a CI step on the new Jenkins server. We could use Sphinx for that. It would update with each new mer

That could be a more easier and direct approach so far?

**Comment by james-prior** [comment]
@TylerReedMC wrote:

> Technologies.pdf
> FMS Requirements.pdf
> HAPI Enclosure Requirements.pdf

Thanks for the dox. It's great to see progress on the dox.

The [Unix philosophy](https://en.wikipedia.org/wiki/Unix_philosophy#Mike_Gancarz:_The_UNIX_Philosophy) loves plain text.
Please commit the plain text source code for those files.

If you use restructured text (rst) you have plain text files that sphinx
can convert to PDF, HTML, and other formats.
I.e., with a [single source of truth](https://en.wikipedia.org/wiki/Single_source_of_truth),
one creates multiple formats from a single file.
Use [make](https://en.wikipedia.org/wiki/Make_%28software%29) to invoke sphinx
according to a [Makefile](https://en.wikipedia.org/wiki/Makefile).
You might invoke make from your continuous integration stuff.
That would be something to ask Jason Green about tonight.
I forget if sphinx [grok](https://en.wikipedia.org/wiki/Grok)s markdown also.

**Comment by james-prior** [comment]
I created a couple of rst files from Tyler's pdfs.
See pull request #77 and pull request #78.
They can use improvement.

**Comment by james-prior** [comment]
@matthew-dews wrote:

> I'd be happy to do the conversion.

Fantastic!

Of the four PDFs above that Tyler posted, I converted a couple of them to rst.
Feel free to convert the other PDFs to rst.
Also, feel free to improve what I did.
You might want to coordinate with @PedroSFreitas,
since he indicated that he's working on it.

**Comment by PedroSFreitas** [comment]
> since he indicated that he's working on it.

Actually I indicated someone else was working on it, but I didn't assign it to me. 😆 
But It'd be a pleasure to help on that when I have the time.

**Comment by james-prior** [comment]
[pdftotext](https://en.wikipedia.org/wiki/Pdftotext)
was pointed out to me as a handy command line program
for the first part of the conversion to .rst.

**Comment by TylerReedMC** [comment]
Or we could take the original .odt file and save out the text. Would still have to deal with images. I would have dropped them here but github doesn't allow it so I converted to pdf.
